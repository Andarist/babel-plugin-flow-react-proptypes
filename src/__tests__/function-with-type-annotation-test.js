var babel = require('babel-core');
var content = `
type FooProps = {
  name: string,
}

const C = (props: FooProps) : ReactElement => {
}
`;

it('function-with-type-annotation', () => {
  var res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
