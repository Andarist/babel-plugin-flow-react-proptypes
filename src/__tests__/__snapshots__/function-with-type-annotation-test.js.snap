exports[`test function-with-type-annotation 1`] = `
"\"use strict\";

var C = function C(props) {};
C.propTypes = {
  name: require(\"react\").PropTypes.string.isRequired
};"
`;
